src/library/columncache.cpp

```diff
@@ -81,7 +81,7 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GENRE], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_COMPOSER], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GROUPING], sortNoCase);
-    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortInt);
+    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortNoCase);
```

Comment:
What happens if the sorting remains at sortInt = cast(%1 as integer), sqlite does probably the job for us. 

An other thought: I have seen track numbers from vinyl rips, labelled A1 ... B2  
http://www.discogs.com/Rolf-Und-Seine-Freunde-Rolfs-Schulweg-Hitparade/release/4633319

So a final solution would be to register a custom sorter. This can also deal with the disk number field.